src/main/scala/analysis/Cfg.scala

```diff
@@ -51,382 +196,930 @@ trait CfgNode:
+                If this is a direct call, then we simply use that call node as the start of the block.
+                However, GoTos in the CFG are resolved as edges, and so there doesn't exist a node to use as
+                the start. Thus we introduce a "ghost" node to act as that jump point - it has no functionality
+                and will simply be skipped by analyses.
+
+                Currently we display these nodes in the DOT view of the CFG, however these could be hidden if desired.
+           */
+          jmps.head match {
+            case jmp: GoTo =>
+              // `GoTo`s are just edges, so introduce a fake `start of block` that can be jmp'd to
+              val ghostNode = CfgGhostNode(block = block)
+              cfg.addEdge(prevNode, ghostNode, cond)
+              precNode = ghostNode
+              visitedBlocks += (block -> ghostNode)
+            case _ =>
+              // (In)direct call - use this as entrance to block
+              visitedBlocks += (block -> jmpNode)
           }
-        } else {
-          blocks(blockName).data.jumps.foreach {
-            case g: GoTo =>
-              if (newBlocks(g.target.label).isEmpty) {
-                visitBlock(g.target.label)
+        }
+
+        /* Possible `jmp` combinations:
+              1. DirectCall
+              2. IndirectCall
+              3. GoTo
+              4. GoTo + GoTo
+         */
+        jmps.head match {
+          case goto: GoTo =>
+            // Process first jump
+            var targetBlock: Block = goto.target
+            var targetCond: Expr = goto.condition match {
+              case Some(c) => c
+              case None    => TrueLiteral
+            }
+
+            // Jump to target block
+            if (visitedBlocks.contains(targetBlock)) {
+              val targetBlockEntry: CfgCommandNode = visitedBlocks(targetBlock)
+              cfg.addEdge(precNode, targetBlockEntry, targetCond)
+            } else {
+              visitBlock(targetBlock, precNode, targetCond)
+            }
+
+            /* If the first GoTo's condition is not `True`, then we know there must be a second conditional jump.
+             *  We could equivalently check if `jmps` has 2 jumps, though this emphasises why we check for
+             *  a second jump. We can assume we have a maximum of 2 jumps, as that's what the IR (as of writing)
+             *  intends.
+             */
+            if (targetCond != TrueLiteral) {
+              val secondGoto: GoTo = jmps.tail.head.asInstanceOf[GoTo]
+              targetBlock = secondGoto.target
+              // IR doesn't store negation of condition, so we must do it manually
+              targetCond = negateConditional(targetCond)
+
+              // Jump to target block
+              if (visitedBlocks.contains(targetBlock)) {
+                val targetBlockEntry: CfgCommandNode = visitedBlocks(targetBlock)
+                cfg.addEdge(precNode, targetBlockEntry, targetCond)
```

Comment:
Maximum of 2 jumps is not a safe assumption. We should try to handle the jumps in a more robust way than this